src/sparseml/pytorch/sparsification/training/modifier_params.py

```diff
@@ -149,6 +149,8 @@ def update(
             for param in self._module_params:
                 self._original.append(param.requires_grad)
                 param.requires_grad = self._trainable
+                if not self._trainable:
+                    param.grad = None  # clear .grad to prevent optimizer updates
```

Comment:
you mind put a short explanation here as to why we need this extra step (that the (Adam) optimizer looks at .grad and ignores requires_grad when updating params)?